src/sentry/integrations/utils/code_mapping.py

```diff
@@ -73,23 +71,35 @@ def stacktrace_buckets(self, stacktraces: List[str]) -> Dict[str, Any]:
                 continue
         return buckets
 
-    def generate_code_mappings(self, stacktraces: List[str]) -> List[CodeMapping]:
-        """Generate code mappings based on the initial trees object and the list of stack traces"""
-        # We need to make sure that calling this method with a new list of stack traces
-        # should always start with a clean slate
-        self.code_mappings = {}
-        buckets: Dict[str, Any] = self.stacktrace_buckets(stacktraces)
-
+    def process_stackframes(self, buckets: Dict[str, List[FrameFilename]]) -> bool:
+        """This processes all stackframes and returns if a new code mapping has been generated"""
+        reprocess = False
         for stackframe_root, stackframes in buckets.items():
             if not self.code_mappings.get(stackframe_root):
                 for frame_filename in stackframes:
                     code_mapping = self._find_code_mapping(frame_filename)
                     if code_mapping:
-                        # XXX: Reprocess feature
-                        # XXX: Reverse mapping may need to be on a per-repo basis
-                        # This helps excluding some source files when we have a matching code mapping
-                        self.reverse_mapping.append(code_mapping.source_path)
+                        # This allows processing some stack frames that
+                        # were matching more than one file
+                        reprocess = True
                         self.code_mappings[stackframe_root] = code_mapping
+        return reprocess
+
+    def generate_code_mappings(self, stacktraces: List[str]) -> List[CodeMapping]:
+        """Generate code mappings based on the initial trees object and the list of stack traces"""
+        # We need to make sure that calling this method with a new list of stack traces
+        # should always start with a clean slate
+        self.code_mappings = {}
+        buckets: Dict[str, List[FrameFilename]] = self.stacktrace_buckets(stacktraces)
+
+        # We reprocess stackframes until we are told that no code mappings were produced
+        # This is order to reprocess past stackframes in light of newly discovered code mappings
+        # This allows for idempotency since the order of the stackframes will not matter
+        # This has no performance issue because stackframes that match an existing code mapping
+        # will be skipped
+        while True:
+            if not self.process_stackframes(buckets):
+                break
```

Comment:
If we introduce a bug in the call, this loop will make the tests run infinitely, thus, we're safe using `while True`.